### ISSUE TITLE
Automatically set parameters of `EwaldPotential` based on given accuracy

### ISSUE THREAD

**Issue by GardevoirX** [lab-cosmo/torch-pme#44]
The current parameter setting method is empirical:

https://github.com/lab-cosmo/torch-pme/blob/6bae97a76136b207f0ff6503d097c36cf8801d2d/src/torchpme/calculators/base.py#L382-L385

https://github.com/lab-cosmo/torch-pme/blob/6bae97a76136b207f0ff6503d097c36cf8801d2d/src/torchpme/calculators/ewaldpotential.py#L54-L55
This may lead to an imbalanced real part error and Fourier part error, impeding the accuracy of the final result. Thus it is necessary to improve this process based on error formulas.

[This paper](https://www2.icp.uni-stuttgart.de/~icp/mediawiki/images/4/4d/Script_Longrange_Interactions.pdf) shows us the following formulas:
![image](https://github.com/user-attachments/assets/73f2db53-cf51-4ffc-adda-8f94046c9e42)
and also a way to search for the best parameters:
![image](https://github.com/user-attachments/assets/be189fb2-ec95-4637-b6cb-d796dbd3f6b4)

Basically, there will be a new method called `from_accuracy`, which may look like:
```python
@classmethod
def from_accuracy(cls, accuracy, ...) -> Tuple["EwaldPotential", float]:
    # Search for the alpha, K and r_c (atomic_smearing, lr_wavelength, and radius cutoff)
    ...
    # Process other parameters
    ...
    # Init the class
    ...
    return potential, r_c
```
The r_c here can be used for the calculation of neighbor-related stuff.
